core/src/serve_repair.rs

```diff
@@ -420,6 +434,20 @@ impl ServeRepair {
                         Self::report_reset_stats(&me, &mut stats);
                         last_print = Instant::now();
                     }
+                    const INTERVAL_MS: u64 = 1000;
+                    const MAX_BYTES_PER_SECOND: usize = 10_000 * 1200;
+                    const MAX_BYTES_PER_INTERVAL: usize =
+                        MAX_BYTES_PER_SECOND * INTERVAL_MS as usize / 1000;
+                    const MAX_BYTES_BUDGET: usize = MAX_BYTES_PER_INTERVAL * 5;
+                    data_budget.update(INTERVAL_MS, |bytes| {
+                        std::cmp::min(
+                            // If there are some leftover bytes from the previous interval, they will
+                            // be added to the `MAX_BYTES_PER_INTERVAL` for the next interval, up to a
+                            // maximum of `MAX_BYTES_BUDGET`.
+                            bytes.saturating_add(MAX_BYTES_PER_INTERVAL),
```

Comment:
Are we ok with 5x overshoot above max-budget just because not exhausting the budget in previous intervals?
I don't see much value in this overshooting logic.

Reply:
The one in cluster-info was added long ago in https://github.com/solana-labs/solana/pull/9550
So it was just a legacy logic carried over to the new implementation.